samza-core/src/main/java/org/apache/samza/processor/JobCoordinatorListener.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.processor;
+
+import org.apache.samza.job.model.JobModel;
+
+/**
+ * Listener interface that can be registered with a {@link org.apache.samza.coordinator.JobCoordinator} instance in order
+ * to receive notifications.
+ */
+public interface JobCoordinatorListener {
+  /**
+   * Method invoked by a {@link org.apache.samza.coordinator.JobCoordinator} in the following scenarios:
```

Comment:
That's not really an expectation from the JC, right? The StreamProcessor cares about stopping any previous job model before returning. It feels confusing to document it here.